securetar/__init__.py

```diff
@@ -36,7 +36,7 @@ class SecureTarFile:
 
     def __init__(
         self,
-        name: Path,
+        name: Path | None,
```

Comment:
We could make this optional and default to `None`, we should then also make the `mode` optional like it is in in stdlib: https://docs.python.org/3/library/tarfile.html#tarfile.TarFile

Reply:
Yeah seems sensible, why not, then we are fully aligned.